files/brews/nginx.rb

```diff
@@ -1,67 +1,206 @@
 require 'formula'
 
 class Nginx < Formula
-  homepage 'https://nginx.org/'
-  url 'https://nginx.org/download/nginx-1.10.2.tar.gz'
-  sha256 '1045ac4987a396e2fa5d0011daf8987b612dd2f05181b67507da68cbe7d765c2'
-  version '1.10.2-boxen1'
+  desc "HTTP(S) server and reverse proxy, and IMAP/POP3 proxy server"
+  homepage "https://nginx.org/"
+  url "https://nginx.org/download/nginx-1.12.1.tar.gz"
+  sha256 "8793bf426485a30f91021b6b945a9fd8a84d87d17b566562c3797aba8fac76fb"
+  version '1.12.1-boxen1'
 
-  depends_on 'pcre'
+  head "http://hg.nginx.org/nginx/", :using => :hg
 
-  skip_clean 'logs'
-
-  def options
-    [
-      ['--with-passenger',   "Compile with support for Phusion Passenger module"],
-      ['--with-webdav',      "Compile with support for WebDAV module"],
-      ['--with-gzip-static', "Compile with support for Gzip Static module"],
-      ['--with-http2',       "Compile with support for the HTTP/2 module"],
-    ]
+  bottle do
+    sha256 "93bcf8e3aec465c219b6c0b4f4d5437c61bf00f2a930ef5702e0521edc51f20e" => :sierra
+    sha256 "8a7c3580534aa0854927f750d4f044a2a85f90d4c1936338a4a09fef7db0824e" => :el_capitan
+    sha256 "0caae754f402abbe1eca413a7f0291fe2499d5779bb1e537d7f80a4d7d3156d3" => :yosemite
   end
 
-  depends_on "pcre"
-  depends_on "openssl" => :recommended
+  devel do
+    url "https://nginx.org/download/nginx-1.13.3.tar.gz"
+    sha256 "5b73f98004c302fb8e4a172abf046d9ce77739a82487e4873b39f9b0dcbb0d72"
+  end
 
-  def passenger_config_args
-      passenger_root = `passenger-config --root`.chomp
+  # Before submitting more options to this formula please check they aren't
+  # already in Homebrew/homebrew-nginx/nginx-full:
+  # https://github.com/Homebrew/homebrew-nginx/blob/master/Formula/nginx-full.rb
+  option "with-passenger", "Compile with support for Phusion Passenger module"
+  option "with-webdav", "Compile with support for WebDAV module"
+  option "with-debug", "Compile with support for debug log"
+  option "with-gunzip", "Compile with support for gunzip module"
 
-      if File.directory?(passenger_root)
-        return "--add-module=#{passenger_root}/ext/nginx"
-      end
+  depends_on "pcre"
+  depends_on "passenger" => :optional
 
-      puts "Unable to install nginx with passenger support. The passenger"
-      puts "gem must be installed and passenger-config must be in your path"
-      puts "in order to continue."
-      exit
+  # passenger uses apr, which uses openssl, so need to keep
+  # crypto library choice consistent throughout the tree.
+  if build.with? "passenger"
+    depends_on "openssl"
+  else
+    depends_on "openssl@1.1"
   end
 
   def install
+    # Changes default port to 8080
+    inreplace "conf/nginx.conf" do |s|
```

Comment:
This block isn't required since we are already managing it via [`templates/config/nginx/nginx.conf.erb`](https://github.com/boxen/puppet-nginx/blob/master/templates/config/nginx/nginx.conf.erb) and if someone wants to change the default, it can be done with hiera.